public/controllers/register-agent/core/register-commands/README.md

```diff
@@ -0,0 +1,304 @@
+# Documentation
+
+- [Register Agent](#register-agent)
+    - [Solution details](#solution-details)
+    - [Configuration details](#configuration-details)
+        - [OS Definitions](#os-definitions)
+            - [Configuration example](#configuration-example)
+            - [Validations](#validations)
+        - [Optional Parameters Configuration](#optional-parameters-configuration)
+            - [Configuration example](#configuration-example-1)
+            - [Validations](#validations-1)
+        - [Command Generator](#command-generator)
+            - [Get install command](#get-install-command)
+            - [Get start command](#get-start-command)
+            - [Get url package](#get-url-package)
+            - [Get all commands](#get-all-commands)
+
+# Register Agent
+
+The register agent is a process that will allow the user to register an agent in the Wazuh Manager. The plugin will provide a form where the user will be able to select the OS and the package that he wants to install. The plugin will generate the registration commands and will show them to the user.
+
+# Solution details
+
+To optimize and make more easier the process to generate the registration commands we have created a class called `Command Generator' that given a set of parameters it will generate the registration commands.
+
+## Configuration
+
+To make the command generator works we need to configure the following parameters and pass them to the class:
+
+## OS Definitions
+
+The OS definitions are a set of parameters that will be used to generate the registration commands. The parameters are the following:
+
+```ts
+export type tOS = 'linux' | 'windows' | 'mac';
+export type tPackageExtensions =
+  | 'rpm'
+  | 'deb'
+  | 'rpm'
+  | 'msi'
+  | 'pkg'
+  | 'p5p'
+  | 'rpm'
+  | 'tar'
+  | 'apk';
```

Comment:
**nitpick**: `rpm` extension is duplicated multiple times.

Reply:
I updated the documentation and the types. Thanks